src/components/App/App.tsx

```diff
@@ -187,10 +197,28 @@ export default class App extends React.Component<AppProps, AppState> {
     this.init();
   }
 
+  getRoomLink = (vanity: string) => {
+    if (vanity) {
+      return `${window.location.origin}/r/${vanity}`;
+    }
+    return `${window.location.origin}${this.state.roomId.replace('/', '#')}`;
+  };
+
+  handleRoomState = (data: any) => {
+    this.setOwner(data.owner);
+    this.setVanity(data.vanity);
+    this.setPassword(data.password);
+    this.setRoomLink(this.getRoomLink(data.vanity));
+    this.setIsChatDisabled(data.isChatDisabled);
+    window.history.replaceState('', '', this.getRoomLink(data.vanity));
+    this.setIsChatDisabled(data.isChatDisabled);
+  };
+
   componentWillUnmount() {
     document.removeEventListener('fullscreenchange', this.onFullScreenChange);
     document.removeEventListener('keydown', this.onKeydown);
     window.clearInterval(this.heartbeat);
+    this.socket.off('REC:getRoomState', this.handleRoomState);
```

Comment:
We don't unmount any of the other handlers--is this special?

Reply:
It's the equivalent of the deleted Effect Hook . https://github.com/howardchung/watchparty/pull/179/files#diff-f2e93f00a5c8ed736b22b40f508386ba658adae42246cb1768443be5f1e6e8ecL67

Reply:
I see. I think the original intent there was to prevent multiple listeners from mounting if the component is mounted/unmounted. However this is not a problem with App.tsx